src/extension.ts

```diff
@@ -117,7 +117,10 @@ export default class RoundedWindowCornersReborn extends Extension {
                 );
                 has_rounded_corners = UI.ShouldHasRoundedCorners(window, cfg);
             }
-            if (!(has_rounded_corners && shadow)) {
+            if (
+                !(has_rounded_corners && shadow) ||
+                window.has_attached_dialogs()
```

Comment:
Since you need to disable the shadow on the attached dialog itself, shouldn't this be `is_attached_dialog()`?

https://gjs-docs.gnome.org/meta14~14/meta.window#method-is_attached_dialog

This would only disable the shadow for the dialog and not for the parent window. I'm not at my laptop right now though, so I can't test that out, lmk if that doesn't work.

Reply:
No, this patch isn't disabling shadow on attached dialog, it disables shadows for `windowPreview` ([src](https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/main/js/ui/windowPreview.js)).

`windowPreview` is a container for each window in the overview. They are responsible for drawing icons and handling DND. Each `windowPreview` contains `Meta.Window` which it represents but attached dialogs don't deserve their own `windowPreview` so they are [added to the parent's `windowPreview`](url).

You might ask "But why we need additional effects for each window preview element in the overview?". The answer is this screenshot.

![Screenshot from 2024-06-28 20-57-21](https://github.com/flexagoon/rounded-window-corners/assets/64484944/ee692f6b-3bad-4f6d-9b4f-ea189fa896f7)

Even if a window actor has a shadow on the desktop, in the overview it's barely visible. And as noted in the comments of the function override, that I edited, since GNOME 43 window actors with effects will become blurry.

Coming back to the bug mentioned in #18, it basically happens, because adding wide or tall attached dialog to the `windowPreview` enlarges it which adds transparent spots on the `windowPreview`. But internally `windowPreview`'s actor is still a rectangular box. Applying shadow effect simply highlights this.